some utils

rhubert opened the pull request:
a collection of tools + libs needed to build another open source project using this layer...

jkloetzke commented on recipes/graphics/xorg/font/font-util.yaml:
```diff
@@ -0,0 +1,45 @@
+inherit: [autotools, autoconf]
+
+metaEnvironment:
+   LICENSE: "MIT + BSD-2-Clause"
+   PKG: "font-util"
+   PKG_VERSION: "1.4.1"
+   PD: "Font utilities dirs"
+   COPYRIGHT: "Copyright (c) 2009, Oracle and/or its affiliates.\n
+               Copyright (c) 2003 The NetBSD Foundation, Inc.\n
+               Copyright (c) 2006 Martin Husemann.\n
+               Copyright (c) 2007 Joerg Sonnenberger.\n
+               Copyright 2005 Red Hat, Inc\n
+               Copyright (c) 1991-2011 Unicode, Inc.  All Rights reserved"
+   LICENSE_FILE: "COPYING"
```
I guess we need to coordinate these meta environment fields. Could you give some rationale for each of the fields (except `PKG_VERSION` of course)? A couple of thoughts:

* shouldn't all variables be prefixed with `PKG_` consistently?
* Is `LICENSE` supposed to be SPDX compatible?
* The `PKG` looks kind of redundant.
* `PD` == package description? How is that used?

rhubert replied:
>     * shouldn't all variables be prefixed with `PKG_` consistently?
> 

Sounds good. 

>     * Is `LICENSE` supposed to be SPDX compatible?
> 
Yes.

>     * The `PKG` looks kind of redundant.
> 
Not sure if we still need this.

>     * `PD` == package description? How is that used?

It's used to build a list of OpenSourceSoftware used in our products. I think it's also included in the SBOM (SoftwareBillOfMaterials) but I'm not 100% sure. For the [CPE-Field](https://en.wikipedia.org/wiki/Common_Platform_Enumeration) of the SBOM we also use `CPE_VENDOR` and `CPE_PRODUCT` if the recipe name is different to generate the correct cpe-string.

The `COPYRIGHT` field is just what `grep -i copyright` found.. Not sure if this should be included in the upstream recipes.

What do you think about the following for upstream recipes?

   | Variable        | Description                           |    
  |-----------------|---------------------------------------|    
  | PKG_LICENSE     | License of the Package SPDX compatible |    
  | PKG_VERSION     | Package Version                       |    
  | PKG_DESCRIPTION | Short description of the package      |    
  | PKG_LICENSE_FILE| Points to a LICENSE file in the src   workspace.                            |    
  | CPE_VENDOR      | Vendor field of CPE schema            |    
  | CPE_PRODUCT     | Product field of CPE schema           |

jkloetzke replied:
Sounds all reasonable. I'm not fully sure about the `CPE_` things, though. But these can be added later anyway.

Regarding `PKG_LICENSE_FILE`. I could imagine there is more than one license file in some packages. How should we handle those?

rhubert replied:
True, this is often used if tools have a different license than the lib. So we could add a different license_file to each multipackage?

jkloetzke commented on recipes/utils/dosfstools.yaml:
```diff
@@ -0,0 +1,21 @@
+
+buildTools: [gettext]
+buildScript: |
+   export ACLOCAL_PATH=${BOB_TOOL_PATHS['gettext']}/../share/aclocal
```
I think this is unnecessary since basement commit 790833a9...

jkloetzke commented on recipes/libs/gcrypt.yaml:
```diff
@@ -0,0 +1,36 @@
+    PV: "1.11.0"
+
+depends:
+    - name: libs::libgpg-error-tool
```
Don't we need a

```yaml
tools:
    target-toolchain: host-compat-toolchain
```

? I cannot see how it works when cross compiling...

jkloetzke commented on classes/xorg-lib.yaml:
```diff
@@ -0,0 +1,17 @@
+
+buildTools: [xorg-util-macros, fontUtil]
+buildSetup: |
+   ACLOCAL_PATH=${BOB_TOOL_PATHS['xorg-util-macros']}/usr/share/aclocal
+   ACLOCAL_PATH+=":${BOB_TOOL_PATHS['fontUtil']}/../share/aclocal"
+   export ACLOCAL_PATH=${ACLOCAL_PATH}
```
Is that really needed at build time? I thought additional automake macros are only required when autoconf and friends are called. :thinking:

jkloetzke commented on recipes/graphics/xorg/font/font-util.yaml:
```diff
@@ -0,0 +1,38 @@
+
+buildTools: [xorg-util-macros]
+buildScript: |
+   export ACLOCAL_PATH=${BOB_TOOL_PATHS['xorg-util-macros']}/usr/share/aclocal
```
Is that really needed for `configure` and `make`?

rhubert replied:
You're right. This is not necessary when using release tarballs. 

We don't have access to them in our build-environment. Instead we're using git-mirrors and have to run `autoconfreconfigure`. As this makes the source-workspace dirty we're using rsync + autoreconf in buildStep and need to set the correct `ACLOCAL_PATH`. That's why it's there...

For up-streaming the recipes I replaced the git-scm with the release tarball and removed only the rsync...

jkloetzke commented on recipes/libs/libxml2.yaml:
```diff
@@ -0,0 +1,37 @@
+   tag: "v${PKG_VERSION}"
+   commit: 3b1742b8391e966be780bdc43fdf959f7b3a118c
+
+buildTools: [target-toolchain, host-toolchain]
```
The `target-toolchain` is redundant.

jkloetzke commented on recipes/libs/libxml2.yaml:
```diff
@@ -0,0 +1,37 @@
+      packageScript: mesonPackageDev
+   tgt:
+      provideDeps: ['*-tgt']
+      packageScript: mesonPackageTgt $1 "!usr/share/gtk-doc" "!*.m4"
```
Excluding `usr/share/gtk-doc` is not required any more since basement commit 32e9ecf5b4205f80dd3043424e57aedc34215054...

jkloetzke commented on recipes/dbus/dbus-glib.yaml:
```diff
@@ -0,0 +1,54 @@
+         - name: dbus::dbus-glib-tools
+           use: [tools]
+           tools:
+              target-toolchain: host-toolchain
```
`host-compat-toolchain`. The `host-toolchain` does not fingerprint and might cause issues with binary packages between different distros.

jkloetzke commented on recipes/dbus/dbus-glib.yaml:
```diff
@@ -0,0 +1,54 @@
+inherit: [autotools, autoconf]
```
Drop `autoconf` class? There is no `checkoutScript` calling `autoconfReconfigure`...

jkloetzke commented on recipes/dbus/dbus.yaml:
```diff
@@ -0,0 +1,61 @@
+
+checkoutDeterministic: True
+checkoutScript: |
+   autoconfReconfigure
```
There is no patch. Why do we need to reconfigure autoconf?

rhubert replied:
There is no configure / Makefile in the release tarball. Only a `configure.ac` ... So I guess `autoconf` is necessary..

jkloetzke commented on recipes/dbus/dbus.yaml:
```diff
@@ -0,0 +1,61 @@
+checkoutScript: |
+   autoconfReconfigure
+
+buildVars: [DEBUG_DBUS]
```
Could we call this `CONFIG_DBUS_DEBUG`? I think having a common schema like `CONFIG_${PKG_NAME}_*` would be beneficial as the number of such options grows...

jkloetzke reviewed (commented):
One general thing: please include the package version in the commit message when adding new recipes...

jkloetzke commented on recipes/devel/mtools.yaml:
This should go to the `utils` namespace. These tools are not confined to software development.

jkloetzke commented on recipes/devel/mtools.yaml:
```diff
@@ -0,0 +1,24 @@
+inherit: [autotools]
+
+metaEnvironment:
+   PKG_VERSION: "4.0.36"
```
Could you please merge this with https://github.com/BobBuildTool/basement-gnu-linux/pull/32/commits/249eb25978e031f282bef3bd6598b43ec60e23d5? I'm also fine with merging @sixtyfourktec's version if you prefer that. (not sure if the libiconv dependency is crucial)

jkloetzke commented on recipes/utils/dosfstools.yaml:
```diff
@@ -0,0 +1,20 @@
+inherit: [autotools]
+
+metaEnvironment:
+   PKG_VERSION: "4.2"
```
Could you add `PKG_LICENSE: "GPL-3.0+`?

jkloetzke commented on recipes/net/libnl.yaml:
It's just a library -> `recipes/libs/libnl.yaml`...

jkloetzke commented on recipes/dbus/dbus-glib.yaml:
At it's core it's a library, even though it comes with a development tool -> `recipes/libs/dbus-glib.yaml`.

jkloetzke commented on recipes/dbus/dbus.yaml:
I think this should go to the `core` namespace. I would not create new namespaces for such daemons...

jkloetzke commented on recipes/net/iptables.yaml:
```diff
@@ -0,0 +1,24 @@
+depends:
+  - kernel::linux-libc-headers
+
+buildVars: [AR, CC]
```
That looks odd. An autotools package usually cross compiles fine. Could you please double check if this is really required?

jkloetzke commented on recipes/net/iptables.yaml:
```diff
@@ -0,0 +1,24 @@
+buildVars: [AR, CC]
+buildTools: [bison, flex]
+buildScript: |
+  CFLAGS="${CFLAGS} -I$1"
```
It doesn't find it's own headers? :astonished:

jkloetzke commented on recipes/net/ethtool.yaml:
```diff
@@ -0,0 +1,27 @@
+
+checkoutDeterministic: True
+checkoutScript: |
+  autoconfReconfigure
```
Do we really need to run `autoreconf` on a vanilla release tarball?

jkloetzke commented on recipes/libs/libxcrypt.yaml:
```diff
@@ -0,0 +1,23 @@
+
+checkoutDeterministic: True
+checkoutScript: |
+  autoconfReconfigure
```
Do we really need to run `autoreconf` on a vanilla release tarball?

rhubert commented:
Thanks for the review. I removed the packages which in the meantime also became part of https://github.com/BobBuildTool/basement-gnu-linux/pull/26 and https://github.com/BobBuildTool/basement-gnu-linux/pull/32 and dropped some other packages that are no longer required ATM.

jkloetzke commented:
Thanks for the quick fixes!
